src/main/java/com/google/devtools/build/lib/actions/AbstractAction.java

```diff
@@ -497,6 +497,12 @@ public static void deleteOutput(Path path, @Nullable Root root) throws IOExcepti
         // Retry deleting after making the parent writable.
         parentDir.setWritable(true);
         deleteOutput(path, root);
```

Comment:
This recursion call here is really confusing and unneccessary. I think we can just replace it with:
```
if (path.isDirectory(Symlinks.NOFOLLOW)) {
  path.deleteTree();
} else {
  path.delete();
}
```

Then we don't need the special check here.

Reply:
Done